Observer/Order/Save.php

```diff
@@ -148,6 +152,23 @@ public function execute(Observer $observer)
                 if ($nostoCustomer instanceof NostoCustomer) {
                     $nostoCustomerId = $nostoCustomer->getNostoId();
                 }
+                // If the id is still null, fetch the `customer_reference`
+                if ($nostoCustomerId === null) {
+                    $customerId = $order->getCustomerId();
+                    try {
+                        $magentoCustomer = $this->magentoCustomerRepository->getById($customerId);
+                    } catch (\Exception $e) {
+                        $this->logger->exception($e);
+                    }
+
+                    // Get the value of `customer_reference`
+                    $customerReferenceAttribute = $magentoCustomer->getCustomAttribute(
```

Comment:
If the exception is caught, this will throw a null pointer error since `$magentoCustomer` hasn't been defined.